### PR TITLE
Delete account

### PR DESCRIPTION
**Pull request by Kuriv2001**
**This PR will allow a user to delete his account:**

The changes that have been made for this are the following:

- A **button** has been added to the **AccountSettingsScreen** (as an action button) that allows a user to delete their account. When clicked the user is prompted if they are sure and only then the account gets deleted => calls the deleteAccount method from the accountViewModel.
- The **deleteAccount** method in the **accountViewModel**: deletes the related event via the eventManager function deleteAllRelatedEvents, deletes the account of the user from the database via **deleteAccount** from **accountManager** and logs out the user from the event an sends him back to the loginScreen.
- The **deleteAllRelatedEvents** method of the **ChimpagneEventManager**: Queries all the events the user has created and that they are part of (as guest/staff). It then deletes the event they created and removes them from the events they joined.

A little video showcasing deleting the account:

https://github.com/Chimpagne/ChimpagneApp/assets/79521990/f66ce7c7-7534-4a7b-a1e5-328cb527ad84

On top:

- The ability to create an event has been restored (was bugged since last merge)
- IconTextButton now also accepts a color for the text, not just the background (allow error color for text)

### CODE REVIEW

File: app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneAccountManager.kt [hunk 1]
```diff
@@ -68,6 +68,14 @@ class ChimpagneAccountManager(
         .addOnFailureListener { onFailure(it) }
   }
 
+  fun deleteAccount(userUid: String, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneAccountManager.kt:71]: Please use the ChimpagneAccountUID type to represent user UIDs

**Review by Athanaze — APPROVED** [review]
This PR achieves the goal of being able to the delete the account.
This will help us when doing manual testing, and makes Chimpagne more GDPR-friendly, in particular regarding this article : https://gdpr-info.eu/art-17-gdpr/ 🥳
The small change mentionned before is necessary to have a coherent codebase.

File: app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneEventManager.kt [hunk 2]
```diff
@@ -220,4 +221,71 @@ class ChimpagneEventManager(
...
+                  updatedStaffs.remove(userUID)
+
+                  events
+                      .document(document.id)
+                      .update(mapOf("guests" to updatedGuests, "staffs" to updatedStaffs))
+                }
+
+            val allTasks = deleteTasks + updateTasks
+            for (task in allTasks) {
+              Log.d("ALL TASKS", task.toString())
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneEventManager.kt:271]: should be removed

File: app/src/main/java/com/monkeyteam/chimpagne/ui/account/AccountSettingsScreen.kt [hunk 3]
```diff
@@ -38,15 +46,37 @@ fun AccountSettingsScreen(
...
+              onFailure = { Log.e("AccountSettingsScreen", "Failed to delete account", it) })
+        },
+        onDismiss = { showDialog = false })
+  }
   Scaffold(
       topBar = {
-        TopBar(
-            text = stringResource(id = R.string.account_settings_screen_title),
-            navigationIcon = { GoBackButton(onClick = onGoBack) })
+        TopAppBar(
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/account/AccountSettingsScreen.kt:65]: please use `com.monkeyteam.chimpagne.ui.components.TopBar`

File: app/src/main/java/com/monkeyteam/chimpagne/ui/account/AccountSettingsScreen.kt [hunk 4]
```diff
@@ -97,3 +127,33 @@ fun SettingItem(label: String, value: String, modifier: Modifier) {
     Text(text = value, style = TextStyle(fontSize = 23.sp, fontFamily = ChimpagneFontFamily))
   }
 }
+
+@Composable
+fun DeleteAccountDialog(onConfirm: () -> Unit, onDismiss: () -> Unit) {
+  androidx.compose.material3.AlertDialog(
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/account/AccountSettingsScreen.kt:133]: should be imported

File: app/src/main/res/values-fr/strings.xml [hunk 5]
```diff
@@ -166,6 +167,10 @@
     <string name="events_near_you">Evénements proches de vous</string>
     <string name="turn_gps_on">Activer votre GPS pour voir les événements proches de vous</string>
     <string name="activate_gps">ACTIVER GPS</string>
+    <string name="confirm_delete_title">Supprimer le compte</string>
+    <string name="confirm_delete_message">Êtes-vous sûr de vouloir supprimer votre compte ?</string>
+    <string name="confirm">Oui</string>
```
Comment (QuicheLorraine3600) [inline app/src/main/res/values-fr/strings.xml:172]: you can use chimpagne_confirm and chimpagne_cancel if you want

**Review by QuicheLorraine3600 — COMMENTED** [review]
Really nice feature, however why did you put the delete button on the top bar ? It makes the title display on 2 line. You should put it above or below the logout button.

File: app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneEventManager.kt (hunk 2 above)
Comment (Kuriv2001) [reply app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneEventManager.kt:271]: Logs have been deleted

File: app/src/main/java/com/monkeyteam/chimpagne/ui/account/AccountSettingsScreen.kt (hunk 3 above)
Comment (Kuriv2001) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/account/AccountSettingsScreen.kt:65]: Has been used

File: app/src/main/java/com/monkeyteam/chimpagne/ui/account/AccountSettingsScreen.kt (hunk 4 above)
Comment (Kuriv2001) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/account/AccountSettingsScreen.kt:133]: is imported

File: app/src/main/res/values-fr/strings.xml (hunk 5 above)
Comment (Kuriv2001) [reply app/src/main/res/values-fr/strings.xml:172]: Used

**Review by QuicheLorraine3600 — APPROVED** [review]

**Pull request merged**